Automattic/wp-calypso
# Editor: Remove global state placeholder actions

This pull request seeks to effectively revert #3025 and #3202, removing placeholder actions from being called in the post editor, for the following reasons:

- The intended direction for editing posts via global Redux state has changed dramatically since the original placeholders were added:
 - Action creators now exist in `state/posts/actions`, not `state/ui/editor/actions` ([source](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/state/posts/actions.js#L132-L148))
 - When editing a post, one must specify the site ID and optional post ID to the action creator ([source](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/state/posts/actions.js#L132-L148))
 - We may not want specialized action creators per field
 - We were premature in making assumptions about how Redux state would be structured, e.g. we can refactor how [we identify whether editing a new post](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/state/ui/editor/selectors.js#L22-L30) in place of our previous approach in [explicitly dispatching actions for editing sessions of new or existing posts](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/post-editor/controller.js#L166-L172)
- Dispatching placeholder actions has a negative effect on performance
 - Dispatching to the global store, even if there's no effective change in state, causes all mounted `connect` components to re-run their state selector logic
 - Subtle issues like [`getContent`](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/post-editor/post-editor.jsx#L554-L557) (a very slow function) being called multiple times
- Inconsistently applied in newer refactorings ([example](https://github.com/Automattic/wp-calypso/blob/fff2769b191af16445d853b3eefd22eb35e26337/client/post-editor/editor-featured-image/index.jsx#L55-L69))
- Causes confusion to developers who may not understand the intent of the duplicated actions
- Progress has stalled on fleshing out behavior of placeholder action creators

__Testing instructions:__

Ensure Mocha tests pass:

```
npm run test-client
```

Verify that there are no regressions in all affected files, which is effectively all editor behavior which has an effect on the values saved for a post.

Ensure no remaining references to `state/ui/editor/post` exist.
